src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Conversion.cs

```diff
@@ -614,6 +614,32 @@ private static bool IsFloatingPointExpressionOfUnknownPrecision(BoundExpression
                         return _factory.Call(null, createSpan, rewrittenOperand, _factory.Literal(length), useStrictArgumentRefKinds: true);
                     }
 
+                case ConversionKind.ImplicitSpan:
+                    {
+                        var spanType = (NamedTypeSymbol)rewrittenType;
+
+                        WellKnownMember ctorMember;
```

Comment:
Why are we looking for the ctor, rather than looking for the existing implicit operator. I hadn't imagined that we'd actually change our codegen for these cases.

Reply:
Yeah, that's a possibility. I went with calling the constructor because
- we already have a WellKnownMember for it (we don't have one for the implicit operator), so it seems better that the compiler relies on lower number of well-known members,
- in general it seems a constructor is a better API to be used than implicit conversion - in particular we don't currently have any conversion well known member it seems.

But
- we will need to add an implicit conversion well known member for Span->ROS to support Span->ROS conversion, so I guess having both would be more consistent,
- the codegen changes.

I don't know if there are any other considerations. Let me know which option you prefer. Now that I have written it down I think I lean towards what you said, so I'm happy to switch over to that.

Reply:
I'm leaning towards the implicit operator for the reasons you mention: consistency with the code required to convert `Span<T>` to `ReadOnlySpan<T>`, and avoids changing the code that we currently generate. Is there a performance reason to prefer one over the other?

Reply:
> Is there a performance reason to prefer one over the other?

I don't think so, both seem equivalent. The implicit conversion calls the constructor. There's also slightly different codegen, so it might result in different IL, but would need a benchmark to compare.